modules/aruco/src/aruco.cpp

```diff
@@ -137,6 +143,10 @@ bool DetectorParameters::readDetectorParameters(const FileNode& fn, Ptr<Detector
     checkRead |= readParameter(fn["maxErroneousBitsInBorderRate"], params->maxErroneousBitsInBorderRate);
     checkRead |= readParameter(fn["minOtsuStdDev"], params->minOtsuStdDev);
     checkRead |= readParameter(fn["errorCorrectionRate"], params->errorCorrectionRate);
+    // new aruco 3 functionality
+    checkRead |= readParameter(fn["useAruco3Detection"], params->useAruco3Detection);
+    checkRead |= readParameter(fn["minSideLengthCanonicalImg"], params->minSideLengthCanonicalImg);
+    checkRead |= readParameter(fn["minMarkerLengthRatioOriginalImg"], params->minMarkerLengthRatioOriginalImg);
     return checkRead;
```

Comment:
`false` result for old .xml files (from existed opencv versions) is an regression.

Reply:
There is no `false` result for old .xml files, because ` |=` operator is using.

Reply:
Also, test `can_find_diamondmarkers` with read old .xml files was added in prev PR. This test successfully completed.  

Only `&=` operator could add `false` result for old .xml files.